crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -524,7 +522,11 @@ impl MultiThreadedExecutor {
         };
 
         #[cfg(feature = "trace")]
-        let task = task.instrument(task_span);
+        let task = task.instrument(
+            self.system_task_metadata[system_index]
+                .system_task_span
+                .clone(),
+        );
```

Comment:
This `clone` makes me wonder, do we really need to instrument the `Future` as well? Normally this is done because `Future`s can suspend execution during `.await` points and that would mess up spans, but here we are never awaiting and moreover we're already measuring the `run_unsafe` internally, which should account for most of the execution time.

Reply:
The instrumentation here does help measure the additional scheduler overhead, which is thankfully very low right now. However, I do see how this both adds even more profiling overhead and how it might not be all that useful to the typical user. Not sure how to best approach toggling this on or off though.

Reply:
My worry is that the time needed for the instrumentation itself here may be in the same order of magniture as the one it is measuring (in addition to the one already measured inside the `run_unsafe`). That is, it is measuring how much time is needed to run the `catch_unwind`, doing a non-blocking send on a channel and checking if the `catch_unwind` returned an error, all of which are pretty fast.